commown_devices/tests/test_product.py

```diff
@@ -0,0 +1,212 @@
+from odoo.exceptions import UserError
+
+from .common import DeviceAsAServiceTC, add_attributes_to_product, create_config
+
+
+class ProductServiceStorableConfigTC(DeviceAsAServiceTC):
+    def setUp(self, *args, **kwargs):
+        super().setUp(*args, **kwargs)
+
+        self.attribute_color = self.env.ref("product.product_attribute_2")
+        color_values = self.env["product.attribute.value"].search(
+            [("attribute_id", "=", self.attribute_color.id)]
+        )
+        self.color1 = color_values[0]
+        self.color2 = color_values[1]
+
+        self.fp3_service_tmpl = self._create_rental_product("fp3+").product_tmpl_id
+        self.fp3_storable_tmpl = self.storable_product.copy({"name": "fp3+"})
+
+        self.protective_screen = self.env["product.template"].create(
+            {
+                "name": "Protective Screen",
+                "type": "product",
+                "tracking": "none",
+            }
+        )
+
+        self.backcover = self.env["product.template"].create(
+            {
+                "name": "Back Cover",
+                "type": "product",
+                "tracking": "none",
+            }
+        )
+        add_attributes_to_product(
+            self.backcover,
+            self.attribute_color,
+            self.color1 + self.color2,
+        )
+        self.backcover.create_variant_ids()
+        self.backcover1, self.backcover2 = self.backcover.product_variant_ids
+
+        add_attributes_to_product(
+            self.fp3_storable_tmpl,
+            self.attribute_color,
+            color_values,
+        )
+        self.fp3_storable_tmpl.create_variant_ids()
+
+        add_attributes_to_product(
+            self.fp3_service_tmpl,
+            self.attribute_color,
+            self.color1 + self.color2,
+        )
+        self.fp3_service_tmpl.create_variant_ids()
+
+        self.fp3_storable_color1 = self.env["product.product"].search(
+            [
+                ("product_tmpl_id", "=", self.fp3_storable_tmpl.id),
+                ("attribute_value_ids.id", "ilike", self.color1.id),
+            ]
+        )
+        self.fp3_storable_color2 = self.env["product.product"].search(
+            [
+                ("product_tmpl_id", "=", self.fp3_storable_tmpl.id),
+                ("attribute_value_ids.id", "ilike", self.color2.id),
+            ]
+        )
+        self.fp3_service_color2 = self.env["product.product"].search(
+            [
+                ("product_tmpl_id", "=", self.fp3_service_tmpl.id),
+                ("attribute_value_ids.id", "ilike", self.color2.id),
+            ]
+        )
+
+    def test_behavior_on_new_variant(self):
```

Comment:
Changer le nom de la méthode ou ajouter une docstring (affichée en cas de plantage du test), pour qu'on comprenne au 1er coup d'oeil ce qui est testé. Exemple : 
`"""Product service configs must be set on new product variants"""`

Reply:
fait